pkg/operator/controller/dns/controller.go

```diff
@@ -251,17 +251,34 @@ func (r *reconciler) publishRecordToZones(zones []configv1.DNSZone, record *iov1
 		// Only publish the record if the DNSRecord has been modified
 		// (which would mean the target could have changed) or its
 		// status does not indicate that it has already been published.
-		if record.Generation == record.Status.ObservedGeneration && recordIsAlreadyPublishedToZone(record, &zone) {
-			log.Info("skipping zone to which the DNS record is already published", "record", record.Spec, "dnszone", zone)
-			continue
-		}
 
 		condition := iov1.DNSZoneCondition{
 			Status:             string(operatorv1.ConditionUnknown),
 			Type:               iov1.DNSRecordFailedConditionType,
 			LastTransitionTime: metav1.Now(),
 		}
 
+		if record.Generation != record.Status.ObservedGeneration && recordIsAlreadyPublishedToZone(record, &zone) {
+			log.Info("replacing DNS record", "record", record.Spec, "dnszone", zone)
+
+			if err := r.dnsProvider.Replace(record, zone); err != nil {
+				log.Error(err, "failed to replace DNS record to zone", "record", record.Spec, "dnszone", zone)
```

Comment:
```suggestion
				log.Error(err, "failed to replace DNS record in zone", "record", record.Spec, "dnszone", zone)
```
Or just delete " to zone".